jenkins/docker-ecr/Jenkinsfile

```diff
@@ -0,0 +1,152 @@
+lib = library(identifier: 'jenkins@20211123', retriever: legacySCM(scm))
+
+pipeline {
+    options {
+        timeout(time: 3, unit: 'HOURS')
+    }
+    agent none
+    parameters {
+        string(
+                defaultValue: 'https://github.com/opensearch-project/opensearch-build',
+                name: 'DOCKER_BUILD_GIT_REPOSITORY',
+                description: 'The git repository name that contains dockerfiles and the docker build script.',
+                trim: true
+        )
+        string(
+                defaultValue: 'main',
+                name: 'DOCKER_BUILD_GIT_REPOSITORY_REFERENCE',
+                description: 'The git reference (branch/tag/commit_id) of above repository.',
+                trim: true
+        )
+        string(
+                defaultValue: 'bash docker/ci/build-image-multi-arch.sh -v <TAG_NAME> -f <DOCKERFILE PATH>',
+                name: 'DOCKER_BUILD_SCRIPT_WITH_COMMANDS',
+                description: 'The script path of the docker build script, assuming you are already in root dir of DOCKER_BUILD_GIT_REPOSITORY.',
+                trim: true
+        )
+        choice(
+                choices: ['staging', 'production'],
+                name: 'ENVIRONMENT',
+                description: 'Are we pushing docker images to staging (opensearchstaging) or production (opensearchproject) account.'
+        )
+        string(
+                name: 'SOURCE_IMAGE_TAG_IN_STAGING',
+                description: 'Only required when ENVIRONMENT is Production- Staging Image tag (Source to copy the image). Eg: 1.3.0',
+                trim: true
+        )
+        string(
+                name: 'DESTINATION_IMAGE_TAG_IN_PROD',
+                description: 'Only required when ENVIRONMENT is Production - Production Image tag (Destination to copy the image to). Eg: 1.3.0',
+                trim: true
+        )
+    }
+    stages {
+        stage('check-params') {
+            agent none
+            steps {
+                script {
+                    if (env.ENVIRONMENT == 'production' && (params.SOURCE_IMAGE_IN_STAGING == '' || params.DESTINATION_IMAGE_IN_PROD == '')) {
+                        currentBuild.result = 'ABORTED'
+                        error('SOURCE_IMAGE_IN_STAGING parameter or DESTINATION_IMAGE_IN_PROD is incorrect')
+                    }
+                    if (env.ENVIRONMENT == 'staging' && !params.SOURCE_IMAGE_IN_STAGING.contains(":")) {
+
+                    }
+                }
+            }
+        }
+        stage('docker-build-dockerhub') {
+            agent {
+                docker {
+                    label 'Jenkins-Agent-Ubuntu2004-X64-m52xlarge-Docker-Builder'
+                    image 'opensearchstaging/ci-runner:ubuntu2004-x64-docker-buildx0.6.3-qemu5.0-awscli1.22-jdk14'
+                    args '-u root -v /var/run/docker.sock:/var/run/docker.sock'
```

Comment:
We should never run jobs as root, instead we should correct permissions issues on the image so it can be used without this go around.

Reply:
Ideally, yes. However, this is how it's done in the current `docker-build` job. Maybe @peterzhuamazon can shed light as to why?